ipc/observability/src/traces_settings.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use serde::Deserialize;
+use serde_with::serde_as;
+use std::path::PathBuf;
+use std::str::FromStr;
+use tracing_appender::rolling::Rotation;
+use tracing_subscriber::filter::EnvFilter;
+
+#[serde_as]
+#[derive(Debug, Deserialize, Clone, Default)]
+pub enum LogLevel {
+    Off,
+    Error,
+    Warn,
+    Info,
+    Debug,
+    #[default]
+    Trace,
+}
+
+impl LogLevel {
+    pub fn as_str(&self) -> &str {
```

Comment:
Have you explored using `strum` instead?

Reply:
💯